### ISSUE TITLE
Patch comment visibility

### ISSUE THREAD

**Issue by geigerzaehler** [radicle-dev/radicle-upstream#2850]
At the moment, a user only sees comments made by peers the user tracks in a patch’s discussion. This is problematic: A user may see comments that are responses to comments they can’t see which makes it hard to follow a discussion. This also means that users have to ensure that people they address in their comments are tracking them, which in turn requires an out-of-band communication channel.

### Solution and alternative

To address this issue we two possibilities. First, Upstream can replicate comments from all peers available on the canonical seed node of a project. Alternatively, we can extend the event log system that underlies comments so that an event references all events its authoring peer replicated. This, in turn, results in peers seeing all comments that that were visible to an author when the author created the comment. This behavior should ensure that every user sees the full context of a discussion.

While the latter approach is more flexible and does not depend on a single seed node, it is also harder to implement and still contains barriers for people joining the network. Therefore replicating from the seed node is our preferred approach.

### Implementation

To have upstream replicate all comments from all peers for a certain patch we need to adjust the `git_fetch` service in the proxy. Given a project and patch it should be possible to construct a ref spec for `git fetch` that will fetch all event logs of all peers for this patch stored on the seed.

**Comment by Kaihuang724** [comment]
I agree that the first solution (replicating comments from all peers available on the seed node) is the way to go. It seems like the cleaner implementation for the time being, although we should revisit this in the future once we have "full p2p"
